# Better DNS syntax highlighting.

--- vim63/runtime/syntax/dns.vim-old	2004-09-26 22:46:26.000000000 -0700
+++ vim63/runtime/syntax/dns.vim	2004-09-26 23:17:20.000000000 -0700
@@ -13,13 +13,14 @@
 endif
 
 " Last match is taken!
-syn match	dnsKeyword	"\<\(IN\|A\|SOA\|NS\|CNAME\|MX\|PTR\|SOA\|MB\|MG\|MR\|NULL\|WKS\|HINFO\|TXT\|CS\|CH\|CPU\|OS\)\>"
+syn match	dnsKeyword	"\<\(IN\|A\|AAAA\|SOA\|NS\|CNAME\|MX\|PTR\|SOA\|MB\|MG\|MR\|NULL\|WKS\|HINFO\|TXT\|CS\|CH\|CPU\|OS\|SRV\)\>"
 syn match   dnsRecordName       "^[^ 	]*"
 syn match   dnsPreProc		"^\$[^ ]*"
 syn match   dnsComment		";.*$"
 syn match   dnsDataFQDN		"\<[^ 	]*\.[ 	]*$"
 syn match   dnsConstant			"\<\([0-9][0-9.]*\|[0-9.]*[0-9]\)\>"
 syn match   dnsIPaddr		"\<\(\([0-2]\)\{0,1}\([0-9]\)\{1,2}\.\)\{3}\([0-2]\)\{0,1}\([0-9]\)\{1,2}\>[ 	]*$"
+syn match   dnsIP6addr		"\<\([[:xdigit:]]\{0,4}:\)\{2,7}\(\|[[:xdigit:]]\{0,4}\)\>[ 	]*$"
 
 " Define the default highlighting.
 " For version 5.7 and earlier: only when not done already
@@ -38,6 +39,7 @@
     HiLink dnsKeyword     Keyword
     HiLink dnsRecordName  Type
     HiLink dnsIPaddr      Type
+    HiLink dnsIP6addr     Type
     HiLink dnsIPerr       Error
     HiLink dnsConstant	  Constant
 
