# TERM=screen.linux background detection, Gentoo bug 83416.
Index: src/option.c
===================================================================
--- src/option.c	(revision 2)
+++ src/option.c	(revision 3)
@@ -3109,11 +3109,15 @@
 	/*
 	 * If 'background' wasn't set by the user, try guessing the value,
 	 * depending on the terminal name.  Only need to check for terminals
-	 * with a dark background, that can handle color.  Only "linux"
-	 * console at the moment.
+	 * with a dark background, that can handle color.  We recognise:
+	 *   * 'linux'           -- linux console
+	 *   * 'screen.linux'    -- linux console, with screen
 	 */
 	idx4 = findoption((char_u *)"bg");
-	if (!(options[idx4].flags & P_WAS_SET) && STRCMP(T_NAME, "linux") == 0)
+	if (!(options[idx4].flags & P_WAS_SET) && (
+		    (STRCMP(T_NAME, "linux") == 0) ||
+		    (STRCMP(T_NAME, "screen.linux") == 0)
+		))
 	{
 	    set_string_option_direct(NULL, idx4, (char_u *)"dark", OPT_FREE);
 	    /* don't mark it as set, when starting the GUI it may be changed
