diff --git a/runtime/syntax/automake.vim b/runtime/syntax/automake.vim
index 2f1ad98..946e833 100644
--- a/runtime/syntax/automake.vim
+++ b/runtime/syntax/automake.vim
@@ -53,6 +53,9 @@ syn region  automakeMakeSString start=+'+  skip=+\\'+  end=+'+  contains=makeIde
 syn region  automakeMakeBString start=+`+  skip=+\\`+  end=+`+  contains=makeIdent,makeSString,makeDString,makeNextLine,automakeSubstitution
 
 " Define the default highlighting.
+
+syn cluster makeCommandsContains add=autoMakeSubst,automakeMakeError,automakeBadSubst
+
 " Only when an item doesn't have highlighting yet
 
 hi def link automakePrimary     Statement
diff --git a/runtime/syntax/make.vim b/runtime/syntax/make.vim
index a6d8ad4..2023080 100644
--- a/runtime/syntax/make.vim
+++ b/runtime/syntax/make.vim
@@ -91,10 +91,8 @@ elseif get(b:, 'make_flavor', s:make_flavor) == 'gnu'
 endif
 
 syn match makeCommandError "^\s\+\S.*" contained
-syn region makeCommands contained start=";"hs=s+1 start="^\t"
-	\ end="^[^\t#]"me=e-1,re=e-1 end="^$"
-	\ contains=makeCmdNextLine,makeSpecial,makeComment,makeIdent,makePreCondit,makeDefine,makeDString,makeSString
-	\ nextgroup=makeCommandError
+syn region makeCommands start=";"hs=s+1 start="^\t" end="^[^\t#]"me=e-1,re=e-1 end="^$" contained contains=@makeCommandsContains nextgroup=makeCommandError
+syn cluster makeCommandsContains add=makeCmdNextLine,makeSpecial,makeComment,makeIdent,makePreCondit,makeDefine,makeDString,makeSString
 syn match makeCmdNextLine	"\\\n."he=e-1 contained
 
 " some directives
