diff --color -ur vim-8.2.2529.orig/gvim-athena.desktop vim-8.2.2529/gvim-athena.desktop
--- vim-8.2.2529.orig/gvim-athena.desktop	2021-02-17 21:57:03.000000000 +0100
+++ vim-8.2.2529/gvim-athena.desktop	2021-03-02 14:24:51.277064340 +0100
@@ -8,7 +8,7 @@
 Name[ru]=GVim
 Name[sr]=GVim
 Name[tr]=GVim
-Name=GVim
+Name=GVim Athena
 # Translators: This is the Generic Application Name used in the Vim desktop file
 GenericName[de]=Texteditor
 GenericName[eo]=Tekstoredaktilo
Only in vim-8.2.2529: gvim-athena.desktop.orig
diff --color -ur vim-8.2.2529.orig/gvim-gnome.desktop vim-8.2.2529/gvim-gnome.desktop
--- vim-8.2.2529.orig/gvim-gnome.desktop	2021-02-17 21:57:03.000000000 +0100
+++ vim-8.2.2529/gvim-gnome.desktop	2021-03-02 14:24:51.277064340 +0100
@@ -8,7 +8,7 @@
 Name[ru]=GVim
 Name[sr]=GVim
 Name[tr]=GVim
-Name=GVim
+Name=GVim GNOME
 # Translators: This is the Generic Application Name used in the Vim desktop file
 GenericName[de]=Texteditor
 GenericName[eo]=Tekstoredaktilo
Only in vim-8.2.2529: gvim-gnome.desktop.orig
diff --color -ur vim-8.2.2529.orig/gvim-gtk.desktop vim-8.2.2529/gvim-gtk.desktop
--- vim-8.2.2529.orig/gvim-gtk.desktop	2021-02-17 21:57:03.000000000 +0100
+++ vim-8.2.2529/gvim-gtk.desktop	2021-03-02 14:24:51.277064340 +0100
@@ -8,7 +8,7 @@
 Name[ru]=GVim
 Name[sr]=GVim
 Name[tr]=GVim
-Name=GVim
+Name=GVim GTK
 # Translators: This is the Generic Application Name used in the Vim desktop file
 GenericName[de]=Texteditor
 GenericName[eo]=Tekstoredaktilo
Only in vim-8.2.2529: gvim-gtk.desktop.orig
diff --color -ur vim-8.2.2529.orig/gvim-motif.desktop vim-8.2.2529/gvim-motif.desktop
--- vim-8.2.2529.orig/gvim-motif.desktop	2021-02-17 21:57:03.000000000 +0100
+++ vim-8.2.2529/gvim-motif.desktop	2021-03-02 14:24:51.277064340 +0100
@@ -8,7 +8,7 @@
 Name[ru]=GVim
 Name[sr]=GVim
 Name[tr]=GVim
-Name=GVim
+Name=GVim Motif
 # Translators: This is the Generic Application Name used in the Vim desktop file
 GenericName[de]=Texteditor
 GenericName[eo]=Tekstoredaktilo
Only in vim-8.2.2529: gvim-motif.desktop.orig
diff --color -ur vim-8.2.2529.orig/runtime/vim.desktop vim-8.2.2529/runtime/vim.desktop
--- vim-8.2.2529.orig/runtime/vim.desktop	2021-02-17 21:57:03.000000000 +0100
+++ vim-8.2.2529/runtime/vim.desktop	2021-03-02 14:24:51.273731018 +0100
@@ -113,6 +113,6 @@
 Icon[fr]=gvim
 Icon[ru]=gvim
 Icon=gvim
-Categories=Utility;TextEditor;
+Categories=ConsoleOnly;TextEditor;
 StartupNotify=false
 MimeType=text/english;text/plain;text/x-makefile;text/x-c++hdr;text/x-c++src;text/x-chdr;text/x-csrc;text/x-java;text/x-moc;text/x-pascal;text/x-tcl;text/x-tex;application/x-shellscript;text/x-c;text/x-c++;
Only in vim-8.2.2529/runtime: vim.desktop.orig
