--- vim74/runtime/syntax/sudoers.vim	2011-02-25 15:05:44.000000000 +0200
+++ vim74/runtime/syntax/sudoers.vim	2013-12-27 20:00:19.631344071 +0200
@@ -23,7 +23,7 @@
 
 syn keyword sudoersTodo               contained TODO FIXME XXX NOTE
 
-syn region  sudoersComment            display oneline start='#' end='$' contains=sudoersTodo
+syn region  sudoersComment            display oneline start='#' end='$' contains=sudoersTodo,sudoersIncludeDirective
 
 syn keyword sudoersAlias              User_Alias Runas_Alias nextgroup=sudoersUserAlias skipwhite skipnl
 syn keyword sudoersAlias              Host_Alias nextgroup=sudoersHostAlias skipwhite skipnl
@@ -142,6 +142,8 @@
 syn match   sudoersCommandArgsInSpec contained '[^[:space:],:=\\]\+\%(\\[[:space:],:=\\][^[:space:],:=\\]*\)*' nextgroup=sudoersCommandArgsInSpec,@sudoersCmndSpec skipwhite
 syn match   sudoersCommandEmptyInSpec contained '""' nextgroup=@sudoersCmndSpec skipwhite skipnl
 
+syn match   sudoersIncludeDirective '^#include\%(dir\)\=' nextgroup=sudoersStringParameter,sudoersListParameter
+
 syn keyword sudoersDefaultEntry Defaults nextgroup=sudoersDefaultTypeAt,sudoersDefaultTypeColon,sudoersDefaultTypeGreaterThan,@sudoersParameter skipwhite skipnl
 syn match   sudoersDefaultTypeAt          contained '@' nextgroup=@sudoersHost skipwhite skipnl
 syn match   sudoersDefaultTypeColon       contained ':' nextgroup=@sudoersUser skipwhite skipnl
@@ -320,6 +322,7 @@
 hi def link sudoersCommandArgs              String
 hi def link sudoersCommandEmpty             Special
 hi def link sudoersDefaultEntry             Keyword
+hi def link sudoersIncludeDirective         Special
 hi def link sudoersDefaultTypeAt            Special
 hi def link sudoersDefaultTypeColon         Special
 hi def link sudoersDefaultTypeGreaterThan   Special
