--- vim-7.0/runtime/filetype.vim~	2006-03-30 22:10:33.046344554 +0300
+++ vim-7.0/runtime/filetype.vim	2006-03-30 23:14:19.721780355 +0300
@@ -2095,6 +2095,8 @@
 " Z-Shell script
 au BufNewFile,BufRead zsh*,zlog*		call s:StarSetf('zsh')
 
+" Nagios config
+au BufNewFile,BufRead /*etc/nagios/*.cfg,*sample-config/template-object/*.cfg{,.in},/var/lib/nagios/objects.cache set filetype=nagios
 
 " Generic configuration file (check this last, it's just guessing!)
 au BufNewFile,BufRead,StdinReadPost *
