--- vim61/src/configure.in.wiget	Wed Oct  9 11:57:36 2002
+++ vim61/src/configure.in	Wed Oct  9 12:00:05 2002
@@ -1632,9 +1632,9 @@
   pdir='/dev'
 fi
 dnl SCO uses ptyp%d
-AC_EGREP_CPP(yes,
+AC_EGREP_CPP(YES,
 [#ifdef M_UNIX
-   yes;
+   YES
 #endif
 	], ptys=`echo /dev/ptyp??`, ptys=`echo $pdir/pty??`)
 dnl if test -c /dev/ptyp19; then
@@ -2075,7 +2075,7 @@
 fi
 
 dnl Check for dynamic linking loader
-AC_CHECK_HEADER(dlfcn.h, DLL=dlfcn.h, AC_CHECK_HEADER(dl.h, DLL=dl.h))
+AC_CHECK_HEADER(dlfcn.h, DLL=dlfcn.h, [AC_CHECK_HEADER(dl.h, DLL=dl.h)])
 if test x${DLL} = xdlfcn.h; then
   AC_DEFINE(HAVE_DLFCN_H, 1, [ Define if we have dlfcn.h. ])
   AC_MSG_CHECKING([for dlopen()])
