--- vim-7.4.979/runtime/syntax/po.vim~	2015-12-19 16:28:18.000000000 +0200
+++ vim-7.4.979/runtime/syntax/po.vim	2015-12-22 10:39:56.522234440 +0200
@@ -55,6 +55,7 @@
 syn match     poDiffOld '\(^#| "[^{]*+}\|{+[^}]*+}\|{+[^}]*\|"$\)' contained
 syn match     poDiffNew '\(^#| "[^{]*-}\|{-[^}]*-}\|{-[^}]*\|"$\)' contained
 syn match     poCommentDiff "^#|.*$" contains=poDiffOld,poDiffNew
+syn match     poUntranslated /^msgstr\s\+""\n\n/
 
 " Translations (also includes header fields as they appear in a translation msgstr)
 syn region poCommentKDE	  start=+"_: +ms=s+1 end="\\n" end="\"\n^msgstr"me=s-1 contained
@@ -103,6 +104,7 @@
   HiLink poDiffOld           Todo
   HiLink poDiffNew          Special
   HiLink poObsolete         Comment
+  HiLink poUntranslated      Todo
 
   HiLink poStatementMsgid   Statement
   HiLink poStatementMsgstr  Statement
