--- vim70/runtime/filetype.vim~	2006-11-15 15:36:16.000000000 +0200
+++ vim70/runtime/filetype.vim	2006-12-22 20:30:52.336234386 +0200
@@ -2070,7 +2070,7 @@
 	\|endif
 
 " Crontab
-au BufNewFile,BufRead crontab,crontab.*		call s:StarSetf('crontab')
+au BufNewFile,BufRead crontab,crontab.*,*.cron		call s:StarSetf('crontab')
 
 " Debian Sources.list
 au BufNewFile,BufRead /etc/apt/sources.list.d/*	call s:StarSetf('debsources')
