--- src/globals.h.old	Tue Jan  5 20:10:00 1999
+++ src/globals.h	Wed Jan 13 13:54:46 1999
@@ -215,7 +215,9 @@
 /* found "-rv" or "-reverse" in command line args */
 EXTERN int	found_reverse_arg INIT(= FALSE);
 EXTERN char *	font_opt INIT(= NULL);
+#endif
 
+#if defined(USE_GUI) || defined(GPM_MOUSE)
 /*
  * While executing external commands or in Ex mode, should not insert GUI
  * events in the input buffer: Set hold_gui_events to non-zero.
