--- vim70/src/installml.sh~	2006-06-06 14:49:21.000000000 +0200
+++ vim70/src/installml.sh	2006-06-06 14:49:21.000000000 +0200
@@ -55,25 +55,25 @@
    # ex
    if test ! -f $destdir/$exname.1 -a -f $destdir/$vimname.1; then
       echo creating link $destdir/$exname.1
-      cd $destdir; ln -s $vimname.1 $exname.1
+      cd $destdir; echo ".so $vimname.1" > $exname.1
    fi
 
    # view
    if test ! -f $destdir/$viewname.1 -a -f $destdir/$vimname.1; then
       echo creating link $destdir/$viewname.1
-      cd $destdir; ln -s $vimname.1 $viewname.1
+      cd $destdir; echo ".so $vimname.1" > $viewname.1
    fi
 
    # rvim
    if test ! -f $destdir/$rvimname.1 -a -f $destdir/$vimname.1; then
       echo creating link $destdir/$rvimname.1
-      cd $destdir; ln -s $vimname.1 $rvimname.1
+      cd $destdir; echo ".so $vimname.1" > $rvimname.1
    fi
 
    # rview
    if test ! -f $destdir/$rviewname.1 -a -f $destdir/$vimname.1; then
       echo creating link $destdir/$rviewname.1
-      cd $destdir; ln -s $vimname.1 $rviewname.1
+      cd $destdir; echo ".so $vimname.1" > $rviewname.1
    fi
 
    # GUI targets are optional
@@ -81,37 +81,37 @@
       # gvim
       if test ! -f $destdir/$gvimname.1 -a -f $destdir/$vimname.1; then
 	 echo creating link $destdir/$gvimname.1
-	 cd $destdir; ln -s $vimname.1 $gvimname.1
+	 cd $destdir; echo ".so $vimname.1" > $gvimname.1
       fi
 
       # gview
       if test ! -f $destdir/$gviewname.1 -a -f $destdir/$vimname.1; then
 	 echo creating link $destdir/$gviewname.1
-	 cd $destdir; ln -s $vimname.1 $gviewname.1
+	 cd $destdir; echo ".so $vimname.1" > $gviewname.1
       fi
 
       # rgvim
       if test ! -f $destdir/$rgvimname.1 -a -f $destdir/$vimname.1; then
 	 echo creating link $destdir/$rgvimname.1
-	 cd $destdir; ln -s $vimname.1 $rgvimname.1
+	 cd $destdir; echo ".so $vimname.1" > $rgvimname.1
       fi
 
       # rgview
       if test ! -f $destdir/$rgviewname.1 -a -f $destdir/$vimname.1; then
 	 echo creating link $destdir/$rgviewname.1
-	 cd $destdir; ln -s $vimname.1 $rgviewname.1
+	 cd $destdir; echo ".so $vimname.1" > $rgviewname.1
       fi
 
       # gvimdiff
       if test ! -f $destdir/$gvimdiffname.1 -a -f $destdir/$vimdiffname.1; then
 	 echo creating link $destdir/$gvimdiffname.1
-	 cd $destdir; ln -s $vimdiffname.1 $gvimdiffname.1
+	 cd $destdir; echo ".so $vimdiffname.1" > $gvimdiffname.1
       fi
 
       # eview
       if test ! -f $destdir/$eviewname.1 -a -f $destdir/$evimname.1; then
 	 echo creating link $destdir/$eviewname.1
-	 cd $destdir; ln -s $evimname.1 $eviewname.1
+	 cd $destdir; echo ".so $evimname.1" > $eviewname.1
       fi
    fi
 fi
--- vim-8.2.2529/src/Makefile.orig	2021-03-02 13:50:22.236523989 +0100
+++ vim-8.2.2529/src/Makefile	2021-03-02 13:50:34.519785983 +0100
@@ -1565,7 +1565,8 @@
 DEST_MAN_PL = $(DEST_MAN_TOP)/pl$(MAN1DIR)
 DEST_MAN_PL_I = $(DEST_MAN_TOP)/pl.ISO8859-2$(MAN1DIR)
 DEST_MAN_PL_U = $(DEST_MAN_TOP)/pl.UTF-8$(MAN1DIR)
-DEST_MAN_RU = $(DEST_MAN_TOP)/ru.KOI8-R$(MAN1DIR)
+DEST_MAN_RU = $(DEST_MAN_TOP)/ru$(MAN1DIR)
+DEST_MAN_RU_K = $(DEST_MAN_TOP)/ru.KOI8-R$(MAN1DIR)
 DEST_MAN_RU_U = $(DEST_MAN_TOP)/ru.UTF-8$(MAN1DIR)
 DEST_MAN_TR = $(DEST_MAN_TOP)/tr$(MAN1DIR)
 DEST_MAN_TR_I = $(DEST_MAN_TOP)/tr.ISO8859-9$(MAN1DIR)
@@ -2577,6 +2578,7 @@
 	-$(SHELL) ./installman.sh xxd $(DEST_MAN_PL_I) "-pl" $(INSTALLMANARGS)
 	-$(SHELL) ./installman.sh xxd $(DEST_MAN_PL_U) "-pl.UTF-8" $(INSTALLMANARGS)
 	-$(SHELL) ./installman.sh xxd $(DEST_MAN_RU) "-ru" $(INSTALLMANARGS)
+	-$(SHELL) ./installman.sh xxd $(DEST_MAN_RU_K) "-ru" $(INSTALLMANARGS)
 	-$(SHELL) ./installman.sh xxd $(DEST_MAN_RU_U) "-ru.UTF-8" $(INSTALLMANARGS)
 	-$(SHELL) ./installman.sh xxd $(DEST_MAN_TR) "-tr" $(INSTALLMANARGS)
 	-$(SHELL) ./installman.sh xxd $(DEST_MAN_TR_I) "-tr" $(INSTALLMANARGS)
@@ -2602,6 +2604,7 @@
 	-$(SHELL) ./installman.sh install $(DEST_MAN_PL_I) "-pl" $(INSTALLMANARGS)
 	-$(SHELL) ./installman.sh install $(DEST_MAN_PL_U) "-pl.UTF-8" $(INSTALLMANARGS)
 	-$(SHELL) ./installman.sh install $(DEST_MAN_RU) "-ru" $(INSTALLMANARGS)
+	-$(SHELL) ./installman.sh install $(DEST_MAN_RU_K) "-ru" $(INSTALLMANARGS)
 	-$(SHELL) ./installman.sh install $(DEST_MAN_RU_U) "-ru.UTF-8" $(INSTALLMANARGS)
 	-$(SHELL) ./installman.sh install $(DEST_MAN_TR) "-tr" $(INSTALLMANARGS)
 	-$(SHELL) ./installman.sh install $(DEST_MAN_TR_I) "-tr" $(INSTALLMANARGS)
@@ -2641,6 +2644,8 @@
 	-$(SHELL) ./installml.sh install "$(GUI_MAN_TARGETS)" \
 		$(DEST_MAN_RU) $(INSTALLMLARGS)
 	-$(SHELL) ./installml.sh install "$(GUI_MAN_TARGETS)" \
+		$(DEST_MAN_RU_K) $(INSTALLMLARGS)
+	-$(SHELL) ./installml.sh install "$(GUI_MAN_TARGETS)" \
 		$(DEST_MAN_RU_U) $(INSTALLMLARGS)
 	-$(SHELL) ./installml.sh install "$(GUI_MAN_TARGETS)" \
 		$(DEST_MAN_TR) $(INSTALLMLARGS)
@@ -2820,6 +2825,7 @@
 	-$(SHELL) ./installman.sh uninstall $(DEST_MAN_PL_I) "" $(INSTALLMANARGS)
 	-$(SHELL) ./installman.sh uninstall $(DEST_MAN_PL_U) "" $(INSTALLMANARGS)
 	-$(SHELL) ./installman.sh uninstall $(DEST_MAN_RU) "" $(INSTALLMANARGS)
+	-$(SHELL) ./installman.sh uninstall $(DEST_MAN_RU_K) "" $(INSTALLMANARGS)
 	-$(SHELL) ./installman.sh uninstall $(DEST_MAN_RU_U) "" $(INSTALLMANARGS)
 	-$(SHELL) ./installman.sh uninstall $(DEST_MAN_TR) "" $(INSTALLMANARGS)
 	-$(SHELL) ./installman.sh uninstall $(DEST_MAN_TR_I) "" $(INSTALLMANARGS)
@@ -2865,6 +2871,8 @@
 	-$(SHELL) ./installml.sh uninstall "$(GUI_MAN_TARGETS)" \
 		$(DEST_MAN_TR) $(INSTALLMLARGS)
 	-$(SHELL) ./installml.sh uninstall "$(GUI_MAN_TARGETS)" \
+		$(DEST_MAN_RU_K) $(INSTALLMLARGS)
+	-$(SHELL) ./installml.sh uninstall "$(GUI_MAN_TARGETS)" \
 		$(DEST_MAN_TR_I) $(INSTALLMLARGS)
 	-$(SHELL) ./installml.sh uninstall "$(GUI_MAN_TARGETS)" \
 		$(DEST_MAN_TR_U) $(INSTALLMLARGS)
@@ -2895,7 +2895,7 @@
 	-rm -f $(DEST_MAN_IT)/xxd.1 $(DEST_MAN_IT_I)/xxd.1 $(DEST_MAN_IT_U)/xxd.1
 	-rm -f $(DEST_MAN_JA_U)/xxd.1
 	-rm -f $(DEST_MAN_PL)/xxd.1 $(DEST_MAN_PL_I)/xxd.1 $(DEST_MAN_PL_U)/xxd.1
-	-rm -f $(DEST_MAN_RU)/xxd.1 $(DEST_MAN_RU_U)/xxd.1
+	-rm -f $(DEST_MAN_RU)/xxd.1 $(DEST_MAN_RU_U)/xxd.1 $(DEST_MAN_RU_K)/xxd.1
 	-rm -f $(DEST_HELP)/*.txt $(DEST_HELP)/tags $(DEST_HELP)/*.pl
 	-rm -f $(DEST_HELP)/*.??x $(DEST_HELP)/tags-??
 	-rm -f $(SYS_MENU_FILE) $(SYS_SYNMENU_FILE) $(SYS_DELMENU_FILE)

