Index: vim/runtime/ftplugin/ruby.vim
===================================================================
--- vim/runtime/ftplugin/ruby.vim.orig
+++ vim/runtime/ftplugin/ruby.vim
@@ -20,6 +20,10 @@
 let s:cpo_save = &cpo
 set cpo&vim
 
+" Debian patch: use ri for keyword lookup
+setlocal keywordprg=ri
+setlocal iskeyword+=?,!
+
 " Matchit support
 if exists("loaded_matchit") && !exists("b:match_words")
   let b:match_ignorecase = 0
