--- vim72/runtime/filetype.vim~	2008-12-27 04:25:56.000000000 +0200
+++ vim72/runtime/filetype.vim	2009-04-02 19:14:10.972877140 +0300
@@ -1682,7 +1682,7 @@
 " Shell scripts (sh, ksh, bash, bash2, csh); Allow .profile_foo etc.
 " Gentoo ebuilds are actually bash scripts
 au BufNewFile,BufRead .bashrc*,bashrc,bash.bashrc,.bash_profile*,.bash_logout*,*.bash,*.ebuild call SetFileTypeSH("bash")
-au BufNewFile,BufRead /*/bash*completion*/* call SetFileTypeSH("bash")
+au BufNewFile,BufRead /*/bash*completion,/*/bash*completion*/* call SetFileTypeSH("bash")
 au BufNewFile,BufRead .kshrc*,*.ksh call SetFileTypeSH("ksh")
 au BufNewFile,BufRead /etc/profile,.profile*,*.sh,*.env call SetFileTypeSH(getline(1))
 
