--- vim-9.1.1975/runtime/filetype.vim~	2025-12-12 14:08:50.000000000 +0100
+++ vim-9.1.1975/runtime/filetype.vim	2025-12-12 14:09:47.976576857 +0100
@@ -1499,7 +1499,7 @@ au BufNewFile,BufRead [a-zA-Z0-9]*Dict{,
       \|endif
 
 " Pam conf
-au BufNewFile,BufRead */etc/pam.d/*		call s:StarSetf('pamconf')
+au BufNewFile,BufRead */etc/pam.d/*,*.pam,*.pamd		call s:StarSetf('pamconf')
 
 " Pandoc
 au BufNewFile,BufRead,BufFilePost *.pandoc,*.pdk,*.pd,*.pdc	setf pandoc
