--- vim60z/src/screen.c~	Tue Apr 10 15:47:51 2001
+++ vim60z/src/screen.c	Tue Apr 10 15:49:00 2001
@@ -2328,7 +2328,7 @@
 
 #ifndef MSDOS
 	/* Check if the character under the cursor should not be inverted */
-	if (!highlight_match && lnum == curwin->w_cursor.lnum && wp == curwin
+	if (!highlight_match && *T_VI == NUL && lnum == curwin->w_cursor.lnum && wp == curwin
 # ifdef FEAT_GUI
 		&& !gui.in_use
 # endif
