--- ./runtime/filetype.vim	2005-12-18 15:58:56.000000000 +0200
+++ /usr/share/vim/vim64/filetype.vim	2005-12-18 15:03:14.000000000 +0200
@@ -88,6 +88,10 @@
 au BufNewFile,BufRead proftpd.conf*		setf apachestyle
 
 " Apache config file
+au BufNewFile,BufRead /etc/httpd/*.conf	let apache_version="2.2" | setf apache
+au BufNewFile,BufRead .*/SOURCES/apache-*.conf	let apache_version="2.2" | setf apache
+au BufNewFile,BufRead /etc/apache/*.conf	let apache_version="1.3" | setf apache
+au BufNewFile,BufRead .*/SOURCES/apache1-*.conf	let apache_version="1.3" | setf apache
 au BufNewFile,BufRead httpd.conf*,srm.conf*,access.conf*,.htaccess,apache.conf* setf apache
 
 " XA65 MOS6510 cross assembler
