--- vim71/runtime/vimrc_example.vim~	2007-05-13 08:47:25.040494112 +0200
+++ vim71/runtime/vimrc_example.vim	2007-05-13 08:49:53.405346643 +0200
@@ -38,7 +38,7 @@
 map Q gq
 
 " In many terminal emulators the mouse works just fine, thus enable it.
-set mouse=a
+"set mouse=a
 
 " Switch syntax highlighting on, when the terminal has colors
 " Also switch on highlighting the last used search pattern.
@@ -83,5 +83,10 @@
 
 " Convenient command to see the difference between the current buffer and the
 " file it was loaded from, thus the changes you made.
+if v:progname =~! "vi"
 command DiffOrig vert new | set bt=nofile | r # | 0d_ | diffthis
 	 	\ | wincmd p | diffthis
+endif
+
+" ADDED BY PLD.
+" USE :help colorscheme FOR INFO
+"set background=dark
+
