--- vim74/runtime/filetype.vim~	2013-09-06 00:03:38.000000000 +0300
+++ vim74/runtime/filetype.vim	2013-09-06 00:04:54.205952274 +0300
@@ -110,8 +110,8 @@
 au BufNewFile,BufRead proftpd.conf*		call s:StarSetf('apachestyle')
 
 " Apache config file
-au BufNewFile,BufRead .htaccess,*/etc/httpd/*.conf		setf apache
-au BufNewFile,BufRead */etc/apache2/sites-*/*.com		setf apache
+au BufNewFile,BufRead .htaccess,*/etc/httpd/*.conf,apache-*.conf,*-apache.conf	let apache_version="2.2" | setf apache
+au BufNewFile,BufRead */etc/apache/*.conf,apache1-*.conf,*-apache1.conf	let apache_version="1.3" | setf apache
 
 " XA65 MOS6510 cross assembler
 au BufNewFile,BufRead *.a65			setf a65
