diff -Nru vim62.orig/src/misc1.c vim62/src/misc1.c
--- vim62.orig/src/misc1.c	2003-11-11 18:53:52.000000000 +0100
+++ vim62/src/misc1.c	2003-11-11 19:29:32.000000000 +0100
@@ -2220,12 +2220,19 @@
 	return p;
     }
 #else
+first_test:
     p = mch_getenv(name);
     if (p != NULL && *p == NUL)	    // empty is the same as not set
 	p = NULL;
 
     if (p != NULL)
 	return p;
+
+    if (strcmp(name, "HOME_ETC") == 0) {
+	name = "HOME";
+	goto first_test;
+    }
+
 #endif
 
     // handling $VIMRUNTIME and $VIM is below, bail out if it's another name.
--- vim62.orig/src/os_unix.h	2003-11-11 18:53:52.000000000 +0100
+++ vim62/src/os_unix.h	2003-11-11 19:53:54.000000000 +0100
@@ -264,7 +264,7 @@
 # ifdef VMS
 #  define USR_EXRC_FILE "sys$login:.exrc"
 # else
-#  define USR_EXRC_FILE "$HOME/.exrc"
+#  define USR_EXRC_FILE "$HOME_ETC/.exrc"
 # endif
 #endif
 
@@ -279,7 +279,7 @@
 # ifdef VMS
 # define USR_VIMRC_FILE  "sys$login:.vimrc"
 # else
-#  define USR_VIMRC_FILE "$HOME/.vimrc"
+#  define USR_VIMRC_FILE "$HOME_ETC/.vimrc"
 # endif
 #endif
 
@@ -294,7 +294,7 @@
 # ifdef VMS
 #  define USR_GVIMRC_FILE "sys$login:.gvimrc"
 # else
-#  define USR_GVIMRC_FILE "$HOME/.gvimrc"
+#  define USR_GVIMRC_FILE "$HOME_ETC/.gvimrc"
 # endif
 #endif
 
@@ -313,7 +313,7 @@
 #  ifdef VMS
 #   define VIMINFO_FILE  "sys$login:.viminfo"
 #  else
-#   define VIMINFO_FILE "$HOME/.viminfo"
+#   define VIMINFO_FILE "$HOME_ETC/.viminfo"
 #  endif
 # endif
 # if !defined(VIMINFO_FILE2) && defined(OS2)
@@ -362,7 +362,7 @@ typedef struct dsc$descriptor   DESC;
 # ifdef VMS
 #  define DFLT_VDIR    "sys$login:vimfiles/view"
 # else
-#  define DFLT_VDIR    "$HOME/.vim/view"       /* default for 'viewdir' */
+#  define DFLT_VDIR    "$HOME_ETC/.vim/view"       /* default for 'viewdir' */
 # endif
 #endif
 
@@ -371,14 +371,14 @@
 # ifdef RUNTIME_GLOBAL
 #  ifdef RUNTIME_GLOBAL_AFTER
 #   define DFLT_RUNTIMEPATH	"~/.vim," RUNTIME_GLOBAL ",$VIMRUNTIME," RUNTIME_GLOBAL_AFTER ",~/.vim/after"
-#   define CLEAN_RUNTIMEPATH	RUNTIME_GLOBAL ",$VIMRUNTIME," RUNTIME_GLOBAL_AFTER
+#   define CLEAN_RUNTIMEPATH	"$HOME_ETC/.vim/after," RUNTIME_GLOBAL ",$VIMRUNTIME," RUNTIME_GLOBAL_AFTER
 #  else
 #   define DFLT_RUNTIMEPATH	"~/.vim," RUNTIME_GLOBAL ",$VIMRUNTIME," RUNTIME_GLOBAL "/after,~/.vim/after"
-#   define CLEAN_RUNTIMEPATH	RUNTIME_GLOBAL ",$VIMRUNTIME," RUNTIME_GLOBAL "/after"
+#   define CLEAN_RUNTIMEPATH	"$HOME_ETC/.vim," RUNTIME_GLOBAL ",$VIMRUNTIME," RUNTIME_GLOBAL "/after"
 #  endif
 # else
 #  define DFLT_RUNTIMEPATH	"~/.vim,$VIM/vimfiles,$VIMRUNTIME,$VIM/vimfiles/after,~/.vim/after"
-#  define CLEAN_RUNTIMEPATH	"$VIM/vimfiles,$VIMRUNTIME,$VIM/vimfiles/after"
+#  define CLEAN_RUNTIMEPATH	"$HOME_ETC/.vim,$VIM/vimfiles,$VIMRUNTIME,$VIM/vimfiles/after"
 # endif
 #endif
 
