diff -burN vim63.orig/runtime/tools/vimspell.sh vim63/runtime/tools/vimspell.sh
--- vim63.orig/runtime/tools/vimspell.sh	2005-03-19 15:54:53.345907600 +0100
+++ vim63/runtime/tools/vimspell.sh	2005-03-19 15:56:32.960763840 +0100
@@ -13,9 +13,7 @@
 # March 1999
 
 INFILE=$1
-OUTFILE=/tmp/vimspell.$$
-# if you have "tempfile", use the following line
-#OUTFILE=`tempfile`
+OUTFILE=`mktemp -t vimspellXXXXXX` || exit 1
 
 #
 # local spellings
