--- vim61/src/configure.in.orig	Fri Oct 11 23:57:55 2002
+++ vim61/src/configure.in	Sat Oct 12 00:04:37 2002
@@ -12,6 +12,8 @@
 
 dnl Checks for programs.
 AC_PROG_CC	dnl required by almost everything
+AC_CHECK_TOOL(EGREP, egrep)
+test -z "$EGREP" && AC_MSG_ERROR([egrep is required but not found in \$PATH])
 AC_PROG_CPP	dnl required by header file checks
 AC_ISC_POSIX	dnl required by AC_C_CROSS
 AC_PROG_AWK	dnl required for "make html" in ../doc
