--- runtime/filetype.vim~	2004-12-04 00:57:07.000000000 +0200
+++ runtime/filetype.vim	2004-12-15 17:35:10.000000000 +0200
@@ -1216,7 +1216,7 @@
 au BufNewFile,BufRead *.moo			setf moo
 
 " Modconf
-au BufNewFile,BufRead */etc/modules.conf,*/etc/modules,*/etc/conf.modules setf modconf
+au BufNewFile,BufRead */etc/modules.conf,*/etc/modules,*/etc/conf.modules,*/etc/modprobe.conf setf modconf
 
 " MPD is based on XML
 au BufNewFile,BufRead *.mpd			setf xml

