--- vim63/src/Makefile.orig	2005-04-26 18:25:11.000000000 +0200
+++ vim63/src/Makefile	2005-04-26 18:25:57.000000000 +0200
@@ -2208,7 +2208,7 @@
 	-@echo 'char_u *compiled_user = (char_u *)"' | tr -d $(NL) >> $@
 	-@if test -n "$(COMPILEDBY)"; then \
 		echo "$(COMPILEDBY)" | tr -d $(NL) >> $@; \
-		else ((logname) 2>/dev/null || whoami) | tr -d $(NL) >> $@; fi
+		else ( (logname 2>/dev/null) || whoami) | tr -d $(NL) >> $@; fi
 	-@echo '";' >> $@
 	-@echo 'char_u *compiled_sys = (char_u *)"' | tr -d $(NL) >> $@
 	-@if test -z "$(COMPILEDBY)"; then hostname | tr -d $(NL) >> $@; fi
