--- vim70/runtime/syntax/fstab.vim	2006-10-20 17:13:47.556873243 +0300
+++ vim70/runtime/syntax/fstab.vim	2006-10-20 15:09:28.000000000 +0300
@@ -26,7 +26,9 @@
 syn match fsDeviceError /\%([^a-zA-Z0-9_\/#@\-:]\)/ contained
 syn keyword fsDeviceKeyword contained none proc linproc tmpfs devpts sysfs usbfs
 syn keyword fsDeviceKeyword contained LABEL nextgroup=fsDeviceLabel
+syn keyword fsDeviceKeyword contained UUID nextgroup=fsDeviceUUID
 syn match fsDeviceLabel contained /=[^ \t]\+/hs=s+1 contains=fsOperator
+syn match fsDeviceUUID contained /=[^ \t]\+/hs=s+1 contains=fsOperator
 
 " Mount Point
 syn cluster fsMountPointCluster contains=fsMountPointKeyword,fsMountPointError
@@ -200,6 +202,7 @@
 	HiLink fsTypeKeyword Type
 	HiLink fsDeviceKeyword Identifier
 	HiLink fsDeviceLabel String
+	HiLink fsDeviceUUID String
 	HiLink fsFreqPassNumber Number
 
 	if exists('fstab_unknown_fs_errors')
