diff -ur vim-5.3.orig/src/screen.c vim-5.3/src/screen.c
--- vim-5.3.orig/src/screen.c	Sun Aug 23 15:49:32 1998
+++ vim-5.3/src/screen.c	Thu Sep  3 03:45:06 1998
@@ -1085,7 +1085,8 @@
 	}
 #ifndef MSDOS
 	/* Check if the character under the cursor should not be inverted */
-	if (!highlight_match && lnum == curwin->w_cursor.lnum && wp == curwin
+	if (!highlight_match && *T_VI == NUL
+		&& lnum == curwin->w_cursor.lnum && wp == curwin
 # ifdef USE_GUI
 		&& !gui.in_use
 # endif
