--- vim-9.1.1975/runtime/filetype.vim~	2025-12-12 08:52:07.000000000 +0100
+++ vim-9.1.1975/runtime/filetype.vim	2025-12-12 13:54:21.961914854 +0100
@@ -1150,6 +1150,9 @@ au BufNewFile,BufRead *.ttl
 	\   setf teraterm |
 	\ endif
 
+" tests for perl modules
+au BufNewFile,BufRead t/*.t			call s:FTpl()
+
 " TeX
 au BufNewFile,BufRead *.tex			call dist#ft#FTtex()
 au BufNewFile,BufRead texdoc.cnf		setf conf
