Index: vim/runtime/syntax/dosini.vim
===================================================================
--- vim/runtime/syntax/dosini.vim.orig
+++ vim/runtime/syntax/dosini.vim
@@ -17,7 +17,7 @@
 
 syn match  dosiniLabel		"^.\{-}="
 syn region dosiniHeader		start="\[" end="\]"
-syn match  dosiniComment	"^;.*$"
+syn match  dosiniComment	"^[;#].*$"
 
 " Define the default highlighting.
 " For version 5.7 and earlier: only when not done already
