--- vim-9.1.1288/runtime/filetype.vim~	2025-04-09 10:16:06.000000000 +0200
+++ vim-9.1.1288/runtime/filetype.vim	2025-04-09 10:25:41.479151284 +0200
@@ -1823,7 +1823,7 @@ au BufNewFile,BufRead {.,}makepkg.conf
 au BufRead pacman.log*					call s:StarSetf('pacmanlog')
 
 " Pam conf
-au BufNewFile,BufRead */etc/pam.conf			setf pamconf
+au BufNewFile,BufRead */etc/pam.conf,*.pam,*.pamd	setf pamconf
 
 " Pam environment
 au BufNewFile,BufRead pam_env.conf,.pam_environment	setf pamenv
