diff -urN vim60z.org/src/os_unix.h vim60z/src/os_unix.h
--- vim60z.org/src/os_unix.h	Mon Apr  2 19:38:15 2001
+++ vim60z/src/os_unix.h	Mon Apr  2 19:41:59 2001
@@ -183,10 +183,10 @@
  * Unix system-dependent file names
  */
 #ifndef SYS_VIMRC_FILE
-# define SYS_VIMRC_FILE "$VIM/vimrc"
+# define SYS_VIMRC_FILE "/etc/vim/vimrc"
 #endif
 #ifndef SYS_GVIMRC_FILE
-# define SYS_GVIMRC_FILE "$VIM/gvimrc"
+# define SYS_GVIMRC_FILE "/etc/vim/gvimrc"
 #endif
 #ifndef DFLT_HELPFILE
 # define DFLT_HELPFILE	"$VIMRUNTIME/doc/help.txt"
