--- vim-9.1.1975/runtime/filetype.vim~	2025-12-12 14:10:10.000000000 +0100
+++ vim-9.1.1975/runtime/filetype.vim	2025-12-12 14:11:42.881271150 +0100
@@ -1077,6 +1077,9 @@ au BufNewFile,BufRead *.smi
 " Snakemake
 au BufNewFile,BufRead Snakefile,*.smk		setf snakemake
 
+" SNMP MIB files (additional to runtime/autoload/dist/ft.vim)
+au BufNewFile,BufRead */mibs/*.txt              setf mib
+
 " Snort Configuration
 au BufNewFile,BufRead *.hog,snort.conf,vision.conf	setf hog
 au BufNewFile,BufRead *.rules			call dist#ft#FTRules()
