--- vim72/runtime/syntax/mysql.vim~	2007-12-20 06:09:52.000000000 +0200
+++ vim72/runtime/syntax/mysql.vim	2010-04-19 18:51:24.000000000 +0300
@@ -62,7 +62,7 @@
 syn match mysqlNumber            "\<0x[abcdefABCDEF0-9]*\>"
 
 " User variables
-syn match mysqlVariable          "@\a*[A-Za-z0-9]*[._]*[A-Za-z0-9]*"
+syn match mysqlVariable          "@\a*[A-Za-z0-9]*\([._]*[A-Za-z0-9]\)*"
 
 " Comments (c-style, mysql-style and modified sql-style)
 syn region mysqlComment          start="/\*"  end="\*/"
