--- vim74/runtime/vimrc_example.vim~	2013-09-06 00:10:56.000000000 +0300
+++ vim74/runtime/vimrc_example.vim	2013-09-06 00:11:24.756404306 +0300
@@ -31,6 +31,9 @@
 set showcmd		" display incomplete commands
 set incsearch		" do incremental searching
 
+" use compressed docs, see :help gzip-helpfile
+set helpfile=$VIMRUNTIME/doc/help.txt.gz
+
 " For Win32 GUI: remove 't' flag from 'guioptions': no tearoff menu entries
 " let &guioptions = substitute(&guioptions, "t", "", "g")
 
@@ -42,13 +45,13 @@
 inoremap <C-U> <C-G>u<C-U>
 
 " In many terminal emulators the mouse works just fine, thus enable it.
-if has('mouse')
-  set mouse=a
-endif
+"if has('mouse')
+"  set mouse=a
+"endif
 
 " Switch syntax highlighting on, when the terminal has colors
 " Also switch on highlighting the last used search pattern.
-if &t_Co > 2 || has("gui_running")
+if (&t_Co > 2 || has("gui_running")) && filereadable($VIMRUNTIME."/syntax/syntax.vim")
   syntax on
   set hlsearch
 endif
@@ -93,7 +93,15 @@
 " Convenient command to see the difference between the current buffer and the
 " file it was loaded from, thus the changes you made.
 " Only define it when not defined already.
-if !exists(":DiffOrig")
+if v:progname =~! "vi" && !exists(":DiffOrig")
   command DiffOrig vert new | set bt=nofile | r ++edit # | 0d_ | diffthis
 		  \ | wincmd p | diffthis
 endif
+
+" Use :help colorscheme for info
+"set background=dark
+
+" Source a global configuration file if available
+if filereadable("/etc/vim/vimrc.local")
+  source /etc/vim/vimrc.local
+endif
