--- vim-5.7/src/vimtutor	Mon Dec 27 09:27:02 1999
+++ vim-5.7.acme/src/vimtutor	Thu Sep 14 13:40:49 2000
@@ -3,7 +3,11 @@
 # Start Vim on a copy of the tutor file.
 
 # Use Vim to copy the tutor, it knows the value of $VIMRUNTIME
-TUTORCOPY=/tmp/tutor$$
+#
+# Tue Aug 24 12:30:09 EST 1999 - Conectiva
+# - fixed tmp race
+#
+TUTORCOPY=`mktemp ${TMPDIR:-/tmp}/tutorXXXXXX`
 export TUTORCOPY
 vim -u NONE -c 'e $VIMRUNTIME/tutor/tutor' -c 'w! $TUTORCOPY' -c 'q'
 
