--- vim62/runtime/ftplugin/spec.vim.orig	2002-04-10 22:03:47.000000000 +0200
+++ vim62/runtime/ftplugin/spec.vim	2003-07-08 00:12:16.000000000 +0200
@@ -127,6 +127,7 @@
 	endfunction
 endif
 
+if !exists("*s:ParseRpmVars")
 function s:ParseRpmVars(str, strline)
 	let end = -1
 	let ret = ""
@@ -154,6 +155,7 @@
 	endwhile
 	return ret
 endfunction
+endif
 
 " The following lines, along with the macros/matchit.vim plugin,
 " make it easy to navigate the different sections of a spec file
