--- vim-8.2.2976/runtime/filetype.vim~	2021-06-12 10:01:08.000000000 +0200
+++ vim-8.2.2976/runtime/filetype.vim	2021-06-12 10:02:30.456368850 +0200
@@ -2347,6 +2347,9 @@ au BufNewFile,BufRead */gitolite-admin/c
 " GTK RC
 au BufNewFile,BufRead .gtkrc*,gtkrc*		call s:StarSetf('gtkrc')
 
+" GYP (Python syntax)
+au BufNewFile,BufRead *.gyp,*.gypi,DEPS		set filetype=python expandtab tabstop=2 shiftwidth=2
+
 " Jam
 au BufNewFile,BufRead Prl*.*,JAM*.*		call s:StarSetf('jam')
 

