--- vim61/src/configure.in.orig	Sat Mar  9 18:29:18 2002
+++ vim61/src/configure.in	Wed Mar 27 23:14:12 2002
@@ -2035,7 +2035,7 @@
 fi
 
 dnl Check for dynamic linking loader
-AC_CHECK_HEADER(dlfcn.h, DLL=dlfcn.h, AC_CHECK_HEADER(dl.h, DLL=dl.h))
+AC_CHECK_HEADER(dlfcn.h, DLL=dlfcn.h, [AC_CHECK_HEADER(dl.h, DLL=dl.h)])
 if test x${DLL} = xdlfcn.h; then
   AC_DEFINE(HAVE_DLFCN_H, 1, [ Define if we have dlfcn.h. ])
   AC_MSG_CHECKING([for dlopen()])
