--- vim-9.1.0905/runtime/syntax/po.vim~	2024-12-04 20:31:43.000000000 +0100
+++ vim-9.1.0905/runtime/syntax/po.vim	2024-12-04 21:09:52.719975448 +0100
@@ -21,6 +21,7 @@ syn match  poStatementMsgCTxt "^msgctxt"
 syn match  poStatementMsgidplural "^msgid_plural" contained
 syn match  poPluralCaseN "[0-9]" contained
 syn match  poStatementMsgstr "^msgstr\(\[[0-9]\]\)" contains=poPluralCaseN
+syn match  poUntranslated /^msgstr\s\+""\n\n/
 
 " Simple HTML and XML highlighting
 syn match  poHtml "<\_[^<>]\+>" contains=poHtmlTranslatables,poLineBreak
@@ -122,6 +123,7 @@ hi def link poCopyrightUnset    Todo
 hi def link poFlagFuzzy         Todo
 hi def link poFlagFormat        Todo
 hi def link poObsolete         Comment
+hi def link poUntranslated      Todo
 
 hi def link poStatementMsgid   Statement
 hi def link poStatementMsgstr  Statement
