--- vim71/src/vimtutor.orig	2007-07-23 10:39:15.051255564 +0200
+++ vim71/src/vimtutor	2007-07-23 10:41:17.038772376 +0200
@@ -39,23 +39,8 @@
 # remove the copy of the tutor on exit
 trap "rm -rf $TODELETE" 0 1 2 3 9 11 13 15
 
-# Vim could be called "vim" or "vi".  Also check for "vimN", for people who
-# have Vim installed with its version number.
-# We anticipate up to a future Vim 8 version :-).
-seq="vim vim8 vim75 vim74 vim73 vim72 vim71 vim70 vim7 vim6 vi"
-for i in $seq; do
-	testvim=`which $i 2>/dev/null`
-	if test -f "$testvim"; then
-		VIM=$i
-		break
-	fi
-done
-
-# When no Vim version was found fall back to "vim", you'll get an error message
-# below.
-if test -z "$VIM"; then
-	VIM=vim
-fi
+# Vim could be called "vim" or "vi".
+[ -x /usr/bin/vim ] && VIM=vim || VIM=vi
 
 # Use Vim to copy the tutor, it knows the value of $VIMRUNTIME
 # The script tutor.vim tells Vim which file to copy
