--- vim-7.4.2189/runtime/vimrc_example.vim~	2016-08-09 22:14:05.000000000 +0200
+++ vim-7.4.2189/runtime/vimrc_example.vim	2016-08-09 22:20:44.071303413 +0200
@@ -26,7 +23,7 @@ else
   endif
 endif
 
-if &t_Co > 2 || has("gui_running")
+if (&t_Co > 2 || has("gui_running")) && filereadable($VIMRUNTIME."/syntax/syntax.vim")
   " Switch on highlighting the last used search pattern.
   set hlsearch
 endif
@@ -56,3 +53,11 @@ endif " has("autocmd")
 if has('syntax') && has('eval')
   packadd matchit
 endif
+
+" Use :help colorscheme for info
+"set background=dark
+
+" Source a global configuration file if available
+if filereadable("/etc/vim/vimrc.local")
+  source /etc/vim/vimrc.local
+endi
--- vim-7.4.2189/runtime/defaults.vim~	2016-08-09 22:14:05.000000000 +0200
+++ vim-7.4.2189/runtime/defaults.vim	2016-08-09 22:26:29.727124083 +0200
@@ -77,13 +77,13 @@ inoremap <C-U> <C-G>u<C-U>
 " can position the cursor, Visually select and scroll with the mouse.
 " Only xterm can grab the mouse events when using the shift key, for other
 " terminals use ":", select text and press Esc.
-if has('mouse')
-  if &term =~ 'xterm'
-    set mouse=a
-  else
-    set mouse=nvi
-  endif
-endif
+"if has('mouse')
+"  if &term =~ 'xterm'
+"    set mouse=a
+"  else
+"    set mouse=nvi
+"  endif
+"endif
 
 " Switch syntax highlighting on when the terminal has colors or when using the
 " GUI (which always has colors).
@@ -99,7 +99,7 @@ endif " has("autocmd")
 " file it was loaded from, thus the changes you made.
 " Only define it when not defined already.
 " Revert with: ":delcommand DiffOrig".
-if !exists(":DiffOrig")
+if v:progname =~! "vi" && !exists(":DiffOrig")
   command DiffOrig vert new | set bt=nofile | r ++edit # | 0d_ | diffthis
 		  \ | wincmd p | diffthis
 endif
