--- vim-8.2.3570/src/vimtutor.orig	2021-11-02 20:05:13.183015908 +0100
+++ vim-8.2.3570/src/vimtutor	2021-11-02 20:07:12.539682552 +0100
@@ -9,14 +9,12 @@
 # When this fails or no argument was given, it tries using 'v:lang'
 # When that also fails, it uses the English version.
 
-# Vim could be called "vim" or "vi".  Also check for "vimN", for people who
-# have Vim installed with its version number.
-# We anticipate up to a future Vim 8.1 version :-).
-seq="vim vim81 vim80 vim8 vim74 vim73 vim72 vim71 vim70 vim7 vim6 vi"
+# Vim could be called "vim" or "vi".
+seq="/usr/bin/vim.heavy /usr/bin/vim /bin/vi"
 if test "$1" = "-g"; then
     # Try to use the GUI version of Vim if possible, it will fall back
     # on Vim if Gvim is not installed.
-    seq="gvim gvim81 gvim80 gvim8 gvim74 gvim73 gvim72 gvim71 gvim70 gvim7 gvim6 $seq"
+    seq="/usr/bin/gvim /usr/bin/gvim.heavy /usr/bin/gvim.athena /usr/bin/gvim.gnome /usr/bin/gvim.gtk /usr/bin/gvim.motif $seq"
     shift
 fi
 
@@ -53,8 +51,7 @@ export TUTORCOPY
 trap "rm -rf $TODELETE" 0 1 2 3 9 11 13 15
 
 for i in $seq; do
-    testvim=$(which $i 2>/dev/null)
-    if test -f "$testvim"; then
+    if test -x "$i"; then
         VIM=$i
         break
     fi
