--- vim73/src/po/Makefile~	2013-02-22 17:47:17.000000000 +0200
+++ vim73/src/po/Makefile	2013-02-22 17:47:53.395264524 +0200
@@ -18,10 +18,8 @@
 		it \
 		ja \
 		ko \
-		ko.UTF-8 \
 		nb \
 		nl \
-		no \
 		pl \
 		pt_BR \
 		ru \
@@ -27,9 +27,7 @@
 		uk \
 		vi \
 		zh_CN \
-		zh_CN.UTF-8 \
-		zh_TW \
-		zh_TW.UTF-8
+		zh_TW
 
 MOFILES = \
 		af.mo \
@@ -56,10 +54,8 @@
 		it.mo \
 		ja.mo \
 		ko.mo \
-		ko.UTF-8.mo \
 		nb.mo \
 		nl.mo \
-		no.mo \
 		pl.mo \
 		pt_BR.mo \
 		ru.mo \
@@ -53,9 +51,7 @@
 		sv.mo \
 		uk.mo \
 		vi.mo \
-		zh_CN.UTF-8.mo \
 		zh_CN.mo \
-		zh_TW.UTF-8.mo \
 		zh_TW.mo
 
 CONVERTED = \
@@ -90,10 +90,8 @@
 		it.ck \
 		ja.ck \
 		ko.ck \
-		ko.UTF-8.ck \
 		nb.ck \
 		nl.ck \
-		no.ck \
 		pl.ck \
 		pt_BR.ck \
 		ru.ck \
@@ -160,12 +160,6 @@
 
 converted: $(CONVERTED)
 
-# Norwegian/Bokmal: "nb" is an alias for "no".
-# Copying the file is not efficient, but I don't know of another way to make
-# this work.
-nb.po: no.po
-	cp no.po nb.po
-
 # Convert ja.po to create ja.sjis.po.  Requires doubling backslashes in the
 # second byte.  Don't depend on sjiscorr, it should only be compiled when
 # ja.sjis.po is outdated.
