Index: vim/runtime/ftplugin/perl.vim
===================================================================
--- vim/runtime/ftplugin/perl.vim.orig
+++ vim/runtime/ftplugin/perl.vim
@@ -39,6 +39,9 @@
 set isfname+=:
 "setlocal iskeyword=48-57,_,A-Z,a-z,:
 
+" Debian patch: use perldoc for keyword lookup
+setlocal keywordprg=perldoc\ -f
+
 " Set this once, globally.
 if !exists("perlpath")
     if executable("perl")
