--- vim62/src/po/Makefile.orig	2004-04-26 21:47:01.944059736 +0200
+++ vim62/src/po/Makefile	2004-04-26 21:47:28.339047088 +0200
@@ -4,10 +4,10 @@
 # Note: ja.sjis, *.cp1250 and zh_CN.cp936 are only for MS-Windows, they are
 # not installed on Unix
 
-LANGUAGES =	af ca cs de en_GB es fr it ja ko no pl ru sk sv uk zh_TW \
+LANGUAGES =	af cs de en_GB es fr it ja ko no pl sk uk zh_TW \
 		zh_TW.UTF-8 zh_CN zh_CN.UTF-8
-MOFILES =	af.mo ca.mo cs.mo de.mo en_GB.mo es.mo fr.mo it.mo ja.mo \
-		ko.mo no.mo pl.mo ru.mo sk.mo sv.mo uk.mo \
+MOFILES =	af.mo cs.mo de.mo en_GB.mo es.mo fr.mo it.mo ja.mo \
+		ko.mo no.mo pl.mo sk.mo uk.mo \
 		zh_TW.mo zh_TW.UTF-8.mo zh_CN.mo zh_CN.UTF-8.mo
 
 PACKAGE = vim
