--- vim-9.1.0905/runtime/filetype.vim~	2024-12-04 21:10:01.000000000 +0100
+++ vim-9.1.0905/runtime/filetype.vim	2024-12-04 21:11:37.149975444 +0100
@@ -3007,7 +3007,7 @@ au BufNewFile,BufRead *.text,README,LICE
 
 " More Apache style config files
 au BufNewFile,BufRead */etc/proftpd/*.conf*,*/etc/proftpd/conf.*/*	call s:StarSetf('apachestyle')
-au BufNewFile,BufRead proftpd.conf*					call s:StarSetf('apachestyle')
+au BufNewFile,BufRead /etc/ftpd/conf.d/*.conf,proftpd*.conf*		call s:StarSetf('apachestyle')
 
 " More Apache config files
 au BufNewFile,BufRead access.conf*,apache.conf*,apache2.conf*,httpd.conf*,httpd-*.conf*,srm.conf*,proxy-html.conf*	call s:StarSetf('apache')
