--- vim-7.4.2189/runtime/vimrc_example.vim~	2016-08-09 22:27:05.000000000 +0200
+++ vim-7.4.2189/runtime/vimrc_example.vim	2016-08-09 22:28:15.756581654 +0200
@@ -21,9 +21,6 @@ if has("vms")
   set nobackup		" do not keep a backup file, use versions instead
 else
   set backup		" keep a backup file (restore to previous version)
-  if has('persistent_undo')
-    set undofile	" keep an undo file (undo changes after closing)
-  endif
 endif
 
 if (&t_Co > 2 || has("gui_running")) && filereadable($VIMRUNTIME."/syntax/syntax.vim")
